data-prepper-plugins/otel-trace-source/src/main/java/com/amazon/dataprepper/plugins/source/oteltrace/OTelTraceGrpcService.java

```diff
@@ -66,4 +109,54 @@ public void export(ExportTraceServiceRequest request, StreamObserver<ExportTrace
                             .asException());
         }
     }
+
+    private void processRequest(
+            final ExportTraceServiceRequest request, final StreamObserver<ExportTraceServiceResponse> responseObserver) {
+        requestsReceivedCounter.increment();
+        payloadSizeSummary.record(request.getSerializedSize());
+
+        if (Context.current().isCancelled()) {
+            requestTimeoutCounter.increment();
+            responseObserver.onError(Status.CANCELLED.withDescription("Cancelled by client").asRuntimeException());
+            return;
+        }
+
+        Collection<Span> spans;
+
+        try {
+            spans = oTelProtoDecoder.parseExportTraceServiceRequest(request);
+        } catch (Exception e) {
+            LOG.error("Failed to parse the request content [{}] due to:", request, e);
+            badRequestsCounter.increment();
+            responseObserver.onError(Status.INVALID_ARGUMENT.withDescription(e.getMessage()).asException());
+            return;
+        }
+
+        final List<Record<Object>> records = spans.stream().map(span -> new Record<Object>(span)).collect(Collectors.toList());
+
+        try {
```

Comment:
Can this code be easily combined with `processRequestWithoutDecoding`? If not, I don't think it is too important since it will be removed in 2.0.

Reply:
So the buffer API call is different. One is using buffer::write while the other is using buffer::writeAll. I am not inclined to introduce any potential change to ux here for the original record type.